I get the same error as @dit7ya in nixpkgs-review:

```
[nix-shell:~/.cache/nixpkgs-review/pr-192391]$ ./results/openai-whisper/bin/whisper 
Traceback (most recent call last):
  File "/nix/store/c1fkpcpz17mccwk9q586a6y2smlwzsbd-python3.10-whisper-unstable-2022-09-21/bin/.whisper-wrapped", line 6, in <module>
    from whisper.transcribe import cli
  File "/nix/store/lv9cxmbxjmma4advdl6h4ihjppvfymg5-python3.9-whisper-unstable-2022-09-21/lib/python3.9/site-packages/whisper/__init__.py", line 8, in <module>
    import torch
  File "/nix/store/ykam416nsgkwymyrmxy366gr5drlkmyf-python3.9-torch-1.12.1/lib/python3.9/site-packages/torch/__init__.py", line 219, in <module>
    raise ImportError(textwrap.dedent('''
ImportError: Failed to load PyTorch C extensions:
    It appears that PyTorch has loaded the `torch/_C` folder
    of the PyTorch repository rather than the C extensions which
    are expected in the `torch._C` namespace. This can occur when
    using the `install` workflow. e.g.
        $ python setup.py install && python -c "import torch"

    This error can generally be solved using the `develop` workflow
        $ python setup.py develop && python -c "import torch"  # This should succeed
    or by running Python from a different directory.
```

Outside of that shell it works:

```
[davidak@gaming:~]$ ~/.cache/nixpkgs-review/pr-192391/results/openai-whisper/bin/whisper 
The cache for model files in Transformers v4.22.0 has been updated. Migrating your old cache. This is a one-time only operation. You can interrupt this and resume the migration later on by calling `transformers.utils.move_cache()`.
Moving 0 files to the new cache system
0it [00:00, ?it/s]
usage: whisper [-h] [--model {tiny.en,tiny,base.en,base,small.en,small,medium.en,medium,large}] [--device DEVICE] [--output_dir OUTPUT_DIR] [--verbose VERBOSE]
               [--task {transcribe,translate}]
               [--language {af,am,ar,as,az,ba,be,bg,bn,bo,br,bs,ca,cs,cy,da,de,el,en,es,et,eu,fa,fi,fo,fr,gl,gu,ha,haw,hi,hr,ht,hu,hy,id,is,it,iw,ja,jw,ka,kk,km,kn,ko,la,lb,ln,lo,lt,lv,mg,mi,mk,ml,mn,mr,ms,mt,my,ne,nl,nn,no,oc,pa,pl,ps,pt,ro,ru,sa,sd,si,sk,sl,sn,so,sq,sr,su,sv,sw,ta,te,tg,th,tk,tl,tr,tt,uk,ur,uz,vi,yi,yo,zh,afrikaans,albanian,amharic,arabic,armenian,assamese,azerbaijani,bashkir,basque,belarusian,bengali,bosnian,breton,bulgarian,burmese,castilian,catalan,chinese,croatian,czech,danish,dutch,english,estonian,faroese,finnish,flemish,french,galician,georgian,german,greek,gujarati,haitian,haitian creole,hausa,hawaiian,hebrew,hindi,hungarian,icelandic,indonesian,italian,japanese,javanese,kannada,kazakh,khmer,korean,lao,latin,latvian,letzeburgesch,lingala,lithuanian,luxembourgish,macedonian,malagasy,malay,malayalam,maltese,maori,marathi,moldavian,moldovan,mongolian,myanmar,nepali,norwegian,nynorsk,occitan,panjabi,pashto,persian,polish,portuguese,punjabi,pushto,romanian,russian,sanskrit,serbian,shona,sindhi,sinhala,sinhalese,slovak,slovenian,somali,spanish,sundanese,swahili,swedish,tagalog,tajik,tamil,tatar,telugu,thai,tibetan,turkish,turkmen,ukrainian,urdu,uzbek,valencian,vietnamese,welsh,yiddish,yoruba}]
               [--temperature TEMPERATURE] [--best_of BEST_OF] [--beam_size BEAM_SIZE] [--patience PATIENCE] [--length_penalty LENGTH_PENALTY]
               [--suppress_tokens SUPPRESS_TOKENS] [--fp16 FP16] [--temperature_increment_on_fallback TEMPERATURE_INCREMENT_ON_FALLBACK]
               [--compression_ratio_threshold COMPRESSION_RATIO_THRESHOLD] [--logprob_threshold LOGPROB_THRESHOLD] [--no_caption_threshold NO_CAPTION_THRESHOLD]
               audio [audio ...]
whisper: error: the following arguments are required: audio
```

auto-detecting and recognizing german works great, even with bad spoken sentences and bad audio quality. looks great so far

--- a/pkgs/development/python-modules/openai-whisper/default.nix
+++ b/pkgs/development/python-modules/openai-whisper/default.nix
@@ -1,0 +1,64 @@
+{ lib
+, fetchFromGitHub
+, buildPythonPackage
+
+# propagates
+, numpy
+, torch
+, tqdm
+, more-itertools
+, transformers
+, ffmpeg-python
+
+# tests
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "whisper";
+  version = "unstable-2022-09-23";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "openai";
+    repo = pname;
+    rev = "8cf36f3508c9acd341a45eb2364239a3d81458b9";
+    hash = "sha256-2RH8eM/SezqFJltelv5AjQEGpqXm980u57vrlkTEUvQ=";
+  };
+
+  postPatch = ''
+    # Rely on the ffmpeg path already patched into the ffmpeg-python library
+    substituteInPlace whisper/audio.py \
+      --replace 'run(cmd="ffmpeg",' 'run('
+  '';
+
+  propagatedBuildInputs = [
+    numpy
+    torch
+    tqdm
+    more-itertools
+    transformers
+    ffmpeg-python
+  ];
+
+  preCheck = ''
+    export HOME=$TMPDIR
+  '';
+
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  disabledTests = [
+    # requires network access to download models
+    "test_transcribe"
+  ];
+
+  meta = with lib; {
+    description = "General-purpose speech recognition model";
+    homepage = "https://github.com/openai/whisper";
+    license = licenses.mit;
+    maintainers = with maintainers; [ hexa ];
+  };
+}
+

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -16986,6 +16986,8 @@ with pkgs;
 
   openai = with python3Packages; toPythonApplication openai;
 
+  openai-whisper = with python3.pkgs; toPythonApplication openai-whisper;
+
   opengrok = callPackage ../development/tools/misc/opengrok { };
 
   openocd = callPackage ../development/embedded/openocd { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -6378,6 +6378,8 @@ in {
 
   open-meteo = callPackage ../development/python-modules/open-meteo { };
 
+  openai-whisper = callPackage ../development/python-modules/openai-whisper { };
+
   openant = callPackage ../development/python-modules/openant { };
 
   openapi-schema-validator = callPackage ../development/python-modules/openapi-schema-validator { };

